### ISSUE TITLE
MadNLP.jl is not precompiling in julia 1.9

### ISSUE THREAD

**Issue by sathvikbhagavan** [MadNLP/MadNLP.jl#257]
I installed MadNLP.jl in a fresh environment:

```julia
(jl_SrdLOn) pkg> st
Status `/tmp/jl_SrdLOn/Project.toml`
  [2621e9c9] MadNLP v0.6.0
```

Julia 1.9 version information:
```julia
Julia Version 1.9.0-rc2
Commit 72aec423c2a (2023-04-01 10:41 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 64 × AMD EPYC 7513 32-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, znver3)
  Threads: 1 on 64 virtual cores
```

Stack trace:

```julia
julia> using MadNLP
[ Info: Precompiling MadNLP [2621e9c9-9eb4-46b1-8089-e8c72242dfb6]
ERROR: LoadError: UndefVarError: `umf_ctrl` not defined
Stacktrace:
  [1] getproperty(x::Module, f::Symbol)
    @ Base ./Base.jl:31
  [2] top-level scope
    @ ~/.julia/packages/MadNLP/pmJUN/src/LinearSolvers/umfpack.jl:1
  [3] include(mod::Module, _path::String)
    @ Base ./Base.jl:457
  [4] include(x::String)
    @ MadNLP ~/.julia/packages/MadNLP/pmJUN/src/MadNLP.jl:4
  [5] top-level scope
    @ ~/.julia/packages/MadNLP/pmJUN/src/LinearSolvers/linearsolvers.jl:145
  [6] include(mod::Module, _path::String)
    @ Base ./Base.jl:457
  [7] include(x::String)
    @ MadNLP ~/.julia/packages/MadNLP/pmJUN/src/MadNLP.jl:4
  [8] top-level scope
    @ ~/.julia/packages/MadNLP/pmJUN/src/MadNLP.jl:35
  [9] include
    @ ./Base.jl:457 [inlined]
 [10] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
    @ Base ./loading.jl:2012
 [11] top-level scope
    @ stdin:2
in expression starting at /home/sathvikbhagavan/.julia/packages/MadNLP/pmJUN/src/LinearSolvers/umfpack.jl:1
in expression starting at /home/sathvikbhagavan/.julia/packages/MadNLP/pmJUN/src/LinearSolvers/linearsolvers.jl:145
in expression starting at /home/sathvikbhagavan/.julia/packages/MadNLP/pmJUN/src/MadNLP.jl:4
in expression starting at stdin:2
ERROR: Failed to precompile MadNLP [2621e9c9-9eb4-46b1-8089-e8c72242dfb6] to "/home/sathvikbhagavan/.julia/compiled/v1.9/MadNLP/jl_CnPLlm".
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, keep_loaded_modules::Bool)
   @ Base ./loading.jl:2262
 [3] compilecache
   @ ./loading.jl:2129 [inlined]
 [4] _require(pkg::Base.PkgId, env::String)
   @ Base ./loading.jl:1772
 [5] _require_prelocked(uuidkey::Base.PkgId, env::String)
   @ Base ./loading.jl:1623
 [6] macro expansion
   @ ./loading.jl:1611 [inlined]
 [7] macro expansion
   @ ./lock.jl:267 [inlined]
 [8] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1574
```

I checked in julia 1.8 and 1.7 as well, in which it is getting precompiled.

**Comment by sshin23** [comment]
Thanks @sathvikbhagavan, this is resolved in  #261, and a new version will be released shortly

**Comment by sshin23** [comment]
Fixed in the v0.7.0 release
